### PR TITLE
feat: Add tracks and rooms filters on sessions page

### LINKED ISSUE #4852: Scheduler/Sessions/Tracks Page: Add Overview of tracks and rooms and make them into filters
The generated website schedule has a rooms and tracks list on the right sidebar, which can also be used as filters, e.g. here http://2019.fossasia.org/event/schedule.html.

Step 1:
Firstly implement such an overview on the schedule and sessions/tracks page on the right side bar. 
Take out the links and twitter badge on these pages from the sidebar.

Step 2:
In a second step implement the rooms and tracks as filters just like the generated website.

Screenshot from 2020-08-24 00-34-00

### PR DESCRIPTION
**Pull request by maze-runnar**
Fixes #4852

image

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by maze-runnar** [comment]
@iamareebjamal this code in routes/sessions.js is not working for filtering -
`if(params.track) {
      filterOptions.push({
            name : 'track.name',
            op : 'eq',
            val : params.track
      })
    }`

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6072219&bid=19883305)

File: app/controllers/public.js [hunk 1]
```diff
@@ -20,6 +20,15 @@ export default class PublicController extends Controller {
     return this.session.currentRouteName && this.session.currentRouteName !== 'public.cfs.new-session' && this.session.currentRouteName !== 'public.cfs.new-speaker' && this.session.currentRouteName !== 'public.cfs.edit-speaker' && this.session.currentRouteName !== 'public.cfs.edit-session';
   }
 
+  @computed('model.tracks')
+  get getTracks() {
```
Comment (niranjan94) [inline app/controllers/public.js:24]: Codacy found an issue: [Trailing spaces not allowed.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6072219)

**Comment by maze-runnar** [comment]
Screenshot from 2020-08-26 09-21-39

**Comment by iamareebjamal** [comment]
Checking

**Comment by iamareebjamal** [comment]
Filter for track should look like this:

`[{"name":"track", "op":"has","val": {"name": "name", "op":"eq", "val":"Business & Sustainability"}}]`

**Comment by iamareebjamal** [comment]
So,

```js
if (params.track) {
  filterOptions.push({
    name : 'track',
    op : 'has',
    val : {
      name: 'name',
      op: 'eq',
      val: params.track
    }
  })
}
```

**Comment by maze-runnar** [comment]
ezgif com-optimize

**Comment by mariobehling** [comment]
Please also use the colors in the sidebar for the tracks, if possible.

**Comment by mariobehling** [comment]
I see whenever we change a filter or date the page goes up to the top of the page. How can we go to the top of the event page heading instead, e.g. here "Sessions".

**Comment by iamareebjamal** [comment]
@maze-runnar Please make the UI as close to the generator UI, with colors for tracks. Also, the chosen track/room should be highlighted and a `minus` sign should be shown beside `Tracks` and `Rooms` heading if any filter is selected so that the filter can be removed. Like clicking on `All` removes the date filter

**Comment by iamareebjamal** [comment]
@mariobehling A page reload is done when a filter is applied. To control the scroll of the page, we'll have to reimplement a lot of things in the route, so I'll create a separate issue to handle this in future

**Comment by maze-runnar** [comment]
Screenshot from 2020-08-26 14-04-27

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 9
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6072219&bid=19883305)

File: app/routes/public/sessions.js [hunk 2]
```diff
@@ -66,6 +72,31 @@ export default class SessionsRoute extends Route {
       });
     }
 
+    if (params.track) {
+      filterOptions.push({
+        name : 'track',
+        op : 'has',
```
Comment (niranjan94) [inline app/routes/public/sessions.js:78]: Codacy found an issue: [Missing space after key 'op'.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6072219)

File: app/routes/public/sessions.js [hunk 3]
```diff
@@ -66,6 +72,31 @@ export default class SessionsRoute extends Route {
...
+        val : {
+          name: 'name',
+          op: 'eq',
```
Comment (niranjan94) [inline app/routes/public/sessions.js:81]: Codacy found an issue: [Missing space after key 'op'.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6072219)

File: app/routes/public/sessions.js [hunk 4]
```diff
@@ -66,6 +72,31 @@ export default class SessionsRoute extends Route {
...
+          val: params.track
```
Comment (niranjan94) [inline app/routes/public/sessions.js:82]: Codacy found an issue: [Missing space after key 'val'.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6072219)

File: app/routes/public/sessions.js [hunk 5]
```diff
@@ -66,6 +72,31 @@ export default class SessionsRoute extends Route {
       });
     }
 
+    if (params.track) {
+      filterOptions.push({
+        name : 'track',
+        op : 'has',
+        val : {
```
Comment (niranjan94) [inline app/routes/public/sessions.js:79]: Codacy found an issue: [Missing space after key 'val'.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6072219)

File: app/controllers/public.js (hunk 1 above)
Comment (niranjan94) [inline app/controllers/public.js:24]: Codacy found an issue: [Trailing spaces not allowed.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6072219)

File: app/routes/public/sessions.js [hunk 6]
```diff
@@ -66,6 +72,31 @@ export default class SessionsRoute extends Route {
...
+          name: 'name',
+          op: 'eq',
+          val: params.track
+        }
+      })
```
Comment (niranjan94) [inline app/routes/public/sessions.js:84]: Codacy found an issue: [Missing semicolon.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6072219)

File: app/routes/public/sessions.js [hunk 7]
```diff
@@ -66,6 +72,31 @@ export default class SessionsRoute extends Route {
...
+        name : 'microlocation',
+        op : 'has',
+        val : {
+          name: 'name',
+          op: 'eq',
+          val: params.room
+        }
+      })
+    }
+
```
Comment (niranjan94) [inline app/routes/public/sessions.js:98]: Codacy found an issue: [Trailing spaces not allowed.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6072219)

File: app/routes/public/sessions.js [hunk 8]
```diff
@@ -66,6 +72,31 @@ export default class SessionsRoute extends Route {
...
+    if (params.room) {
+      filterOptions.push({
+        name : 'microlocation',
+        op : 'has',
+        val : {
+          name: 'name',
+          op: 'eq',
+          val: params.room
```
Comment (niranjan94) [inline app/routes/public/sessions.js:94]: Codacy found an issue: [Missing space after key 'val'.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6072219)

File: app/routes/public/sessions.js [hunk 9]
```diff
@@ -66,6 +72,31 @@ export default class SessionsRoute extends Route {
       });
     }
 
+    if (params.track) {
+      filterOptions.push({
+        name : 'track',
+        op : 'has',
+        val : {
+          name: 'name',
```
Comment (niranjan94) [inline app/routes/public/sessions.js:80]: Codacy found an issue: [Missing space after key 'name'.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6072219)

**Comment by iamareebjamal** [comment]
Color the circle, not the text. And the minus sign should be single (not against every track) and on the right of the text `Tracks` and `Rooms`

**Comment by maze-runnar** [comment]
@iamareebjamal please review -
Screenshot from 2020-08-26 18-52-00

File: app/routes/public/sessions.js (hunk 4 above)
Comment (niranjan94) [inline app/routes/public/sessions.js:82]: Codacy found an issue: [Missing space after key 'val'.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6072219)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 8
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6072219&bid=19883305)

File: app/routes/public/sessions.js (hunk 3 above)
Comment (niranjan94) [inline app/routes/public/sessions.js:81]: Codacy found an issue: [Missing space after key 'op'.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6072219)

File: app/routes/public/sessions.js (hunk 7 above)
Comment (niranjan94) [inline app/routes/public/sessions.js:98]: Codacy found an issue: [Trailing spaces not allowed.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6072219)

File: app/routes/public/sessions.js (hunk 9 above)
Comment (niranjan94) [inline app/routes/public/sessions.js:80]: Codacy found an issue: [Missing space after key 'name'.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6072219)

File: app/routes/public/sessions.js (hunk 5 above)
Comment (niranjan94) [inline app/routes/public/sessions.js:79]: Codacy found an issue: [Missing space after key 'val'.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6072219)

File: app/routes/public/sessions.js (hunk 2 above)
Comment (niranjan94) [inline app/routes/public/sessions.js:78]: Codacy found an issue: [Missing space after key 'op'.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6072219)

File: app/routes/public/sessions.js (hunk 8 above)
Comment (niranjan94) [inline app/routes/public/sessions.js:94]: Codacy found an issue: [Missing space after key 'val'.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6072219)

File: app/routes/public/sessions.js (hunk 6 above)
Comment (niranjan94) [inline app/routes/public/sessions.js:84]: Codacy found an issue: [Missing semicolon.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6072219)

**Comment by mariobehling** [comment]
* [ ] Do we need the minus sign? What for? I think no need.
* [ ] Please reduce size of Tracks and Rooms heading. On the website generator it is heading 4.
* [ ] Please move the side bar filters a) below the event links and b) above the twitter box.

**Comment by maze-runnar** [comment]
@iamareebjamal please take a look at requirements by @mariobehling. And please tell me final changes I have to make

**Comment by mariobehling** [comment]
@maze-runnar I think we are all aligned to make it as similar as possible to the website generator. The question is just how far can we get in this iteration. Would be great, if you make the last changes mentioned by both of us and then we can also try and see how it works on the live site.

**Comment by maze-runnar** [comment]
> * [ ]  Do we need the minus sign? What for? I think no need.
 To clear the room filters

> * [x]  Please reduce size of Tracks and Rooms heading. On the website generator it is heading 4.
> * [x]  Please move the side bar filters a) below the event links and b) above the twitter box.

File: app/routes/public/sessions.js (hunk 5 above)
Comment (niranjan94) [inline app/routes/public/sessions.js:79]: Codacy found an issue: [Missing space after key 'val'.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6072219)

File: app/routes/public/sessions.js (hunk 7 above)
Comment (niranjan94) [inline app/routes/public/sessions.js:98]: Codacy found an issue: [Trailing spaces not allowed.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6072219)

File: app/routes/public/sessions.js (hunk 6 above)
Comment (niranjan94) [inline app/routes/public/sessions.js:84]: Codacy found an issue: [Missing semicolon.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6072219)

File: app/routes/public/sessions.js (hunk 3 above)
Comment (niranjan94) [inline app/routes/public/sessions.js:81]: Codacy found an issue: [Missing space after key 'op'.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6072219)

File: app/routes/public/sessions.js (hunk 4 above)
Comment (niranjan94) [inline app/routes/public/sessions.js:82]: Codacy found an issue: [Missing space after key 'val'.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6072219)

File: app/routes/public/sessions.js (hunk 9 above)
Comment (niranjan94) [inline app/routes/public/sessions.js:80]: Codacy found an issue: [Missing space after key 'name'.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6072219)

File: app/routes/public/sessions.js (hunk 2 above)
Comment (niranjan94) [inline app/routes/public/sessions.js:78]: Codacy found an issue: [Missing space after key 'op'.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6072219)

**Comment by mariobehling** [comment]
Please change filter text font from blue to standard dark color. I think it is black or dark gray.

**Comment by iamareebjamal** [comment]
I'm polishing it @maze-runnar, please take up the grouping issue

**Comment by iamareebjamal** [comment]
@mariobehling I tried a lot to implement clear filters without the minus icon but it doesn't work for some reason. Going for this for now, will revisit another time

**Pull request merged**